### ISSUE TITLE
Questions about project config after moving to SPM

### ISSUE THREAD

**Issue by yo1995** [Esri/arcgis-runtime-samples-ios#1064]
It was a rush when we upgraded to use SPM in #1062 . Although the project built fine on Simulator, there are still a few config items that I want to make sure set correctly.

1. Should dependencies be empty?

![image](https://user-images.githubusercontent.com/9660181/115815397-bec13400-a3ab-11eb-990a-6ae7fa302257.png)

I think it should be empty, because SPM handles package resolving, and the toolkit is guaranteed to be compiled before we build the sample viewer.

2. Should we still specify `LD_RUNPATH_SEARCH_PATHS`?

![image](https://user-images.githubusercontent.com/9660181/115815755-62124900-a3ac-11eb-80ee-d0e7a43e7e84.png)

I'm not quite sure, but since the binary framework is resolved by SPM, it's likely that we don't need this anymore. I test built with it reset to empty, and it succeeded, so I guess it's OK to remove them?

3. There was an error after archive and before upload to App Store

![last-step](https://user-images.githubusercontent.com/9660181/115815900-bb7a7800-a3ac-11eb-88f9-6c98b2597d25.png)

When I archived the app with the current `main` branch, and tried to upload to App Store via Organizer, it prompts me with the errors above. A Google search gave me the links: 

- https://forums.swift.org/t/swift-package-binary-framework-issue/41922/5
- https://bugs.swift.org/browse/SR-13840

Removing the duplicate `ArcGIS.framework` at `***.xcarchive/Products/Applications/ArcGIS Runtime SDK Samples.app/ArcGIS.framework` solve the issue.

I wonder if we need to follow the comments in that thread, to add a run script build phase to remove it? Or is there another solution to work around this issue?

4. Apple Silicon support

> ITMS-90863: Apple silicon Macs support issue - The app links with libraries that are not present on Mac:
@rpath/ArcGIS.framework/ArcGIS
After you’ve corrected the issues, you can upload a new binary to App Store Connect.

5. But... it crashed right away!

🤯

**Comment by yo1995** [comment]
```
$(BUILT_PRODUCTS_DIR)/$(CONTENTS_FOLDER_PATH)/ArcGIS.framework
```

**Comment by philium** [comment]
> Should dependencies be empty?

Yes.

**Comment by philium** [comment]
> Should we still specify `LD_RUNPATH_SEARCH_PATHS`?

Yes. This is the default value for newly created projects.

**Comment by philium** [comment]
> Removing the duplicate `ArcGIS.framework` at `***.xcarchive/Products/Applications/ArcGIS Runtime SDK Samples.app/ArcGIS.framework` solve the issue.

Which version of Xcode are you using? I'm only seeing one copy of the framework inside the `Frameworks` directory when archiving with Xcode 12.0.

**Comment by yo1995** [comment]
> Which version of Xcode are you using? I'm only seeing one copy of the framework inside the `Frameworks` directory when archiving with Xcode 12.0.

I'm using Xcode 12.4.1 (latest MAS version) and macOS 10.15.7 . 
Just now I used a dummy app to test and it seems to work fine, so I'll try to clear `Derived Data` and do a clean build to see if it still happens... 😕

**Comment by yo1995** [comment]
All mysteries resolved with Big Sur and Xcode 12.5
